interview-questions.md

```diff
@@ -8,36 +8,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+Your answer: If rails db:migrate is not initiated, I will delete the model using RESTful control destroy by defining the method in cohorts_controllrer.rb. Then, I will create another model again with forein_key which reference a primary key of another table. If already migrated, I will copy the database and start new. The name of foriegn key is cohort_id:integer and it is on the student's model.
 
-Researched answer:
+Researched answer: If already migrated, I couldn't figure this out but, the best answer I got is from the syllybus. In active record migraion we can edit the database by running a Rails generate command to create a migration with proper naming conventons that communicate the intent of the changes we are making. Simply follow this format $rails generate migration add_item_to_list. In our case type this command in terminal $rails generate migration add_forien_key_to_student. This will create a new version of the database and update the schema file accordingly. Then editing the file and adding the changes we want with in the "def change" method in migrate folder. It should show something like this. Note sure tho.The migration class in active record has methods which does CRUD operations.
+
+class CreateCohorts < ActiveRecord::Migration
+  def change
+    create_table : Cohort do |t|
+      t.string :students
+      t.timestamps
+    end
+  end
+end
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: You can use as many parameters as you want. Therefore, any of the seven RESTful routes can use parameters. 
 
-Researched answer:
+Researched answer: They all use params.
 
```

Comment:
https://github.com/learn-academy-2022-echo/Syllabus/blob/main/rails/restful-routes-crud.md

Please ensure you are giving as much information as possible while interviewing. Short and concise is not the goal. Thorough and precise is the key.